### PR TITLE
New full name pipe

### PR DESCRIPTION
**Pull request by yhedholm**
A while back you suggested we create a fullName pipe. At the time we were scrambling, so I decided to wait. I now implemented the fullName pipe.

### CODE REVIEW

File: backend/src/main/java/com/google/rolecall/services/CastServices.java [hunk 1]
```diff
@@ -61,7 +61,9 @@ public Cast getCast(Integer id) throws EntityNotFoundException, InvalidParameter
     Cast cast = createNewCast(newCast);
     List<String> warnings = verifySubCasts(cast);
 
+    System.out.printf("AAA\n");
```
Comment (mikzat) [inline backend/src/main/java/com/google/rolecall/services/CastServices.java:64]: let's remove these debugging print outs

File: frontend/rolecall/src/app/cast/cast-editor-v2.component.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export class CastEditorV2 implements OnInit {
     let autoSelectFirst = false;
     this.selectedPiece = piece;
     this.updateFilteredCasts();
-    if (this.selectedPiece && this.selectedPiece.uuid !== piece.uuid) {
```
Comment (mikzat) [inline frontend/rolecall/src/app/cast/cast-editor-v2.component.ts:84]: Is this change required for the new pipe implementation? If not, could it be moved to a separate PR?

File: frontend/rolecall/src/app/common_components/full-name.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import {Pipe, PipeTransform} from '@angular/core';
+import {User} from '../api/user_api.service';
+
+@Pipe({
+  name: 'fullName'
+})
+export class FullNamePipe implements PipeTransform {
+
+  public transform(user: User, ...args: unknown[]): string {
```
Comment (mikzat) [inline frontend/rolecall/src/app/common_components/full-name.pipe.ts:9]: This pipe is not accepting arguments, so remove `...args: unknown[]` and just leave `user: User`

File: frontend/rolecall/src/app/common_components/full-name.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    if (!user) {
```
Comment (mikzat) [inline frontend/rolecall/src/app/common_components/full-name.pipe.ts:10]: The arg type is defined as `user: User`, which means it cannot be null or undefined. Remove this if-check?

**Review by mikzat — CHANGES_REQUESTED** [review]

File: frontend/rolecall/src/app/cast/cast-editor-v2.component.ts (hunk 2 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/cast/cast-editor-v2.component.ts:84]: It was a display bug I discovered while verifying that the pipe worked. It was an easy fix, so I though I could sneak it in.

File: frontend/rolecall/src/app/common_components/full-name.pipe.ts (hunk 3 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/common_components/full-name.pipe.ts:9]: Done.

File: frontend/rolecall/src/app/common_components/full-name.pipe.ts (hunk 4 above)
Comment (yhedholm) [reply frontend/rolecall/src/app/common_components/full-name.pipe.ts:10]: I want the pipe to be null/undefined sturdy, so I changed the signature to transform(user?: User) { ... }

File: backend/src/main/java/com/google/rolecall/services/CastServices.java (hunk 1 above)
Comment (yhedholm) [reply backend/src/main/java/com/google/rolecall/services/CastServices.java:64]: Sorry! I missed this because it is on the server and I didn't think I had to clean it up before doin the PR.

Removed.

File: frontend/rolecall/src/app/common_components/full-name.pipe.ts (hunk 4 above)
Comment (mikzat) [reply frontend/rolecall/src/app/common_components/full-name.pipe.ts:10]: Ok cool, in that case let's use `transform(user: User|undefined|null)`.

There are a couple of issues with using `(user?: User)`
* it is not semantically accurate since the parameter is not actually optional, you always have to pass one into a pipe (even if you pass in undefined or null)
* it does not support potential `null`, since `optional` only means it'll be `User` or `undefined`

Comment (yhedholm) [reply frontend/rolecall/src/app/common_components/full-name.pipe.ts:10]: Done.

**Review by mikzat — APPROVED** [review]
